### ISSUE TITLE
RangeError thrown during performLayout() when using OrientationBuilder

### ISSUE THREAD

**Issue by adamkoch** [madewithfelt/flutter_layout_grid#12]
Note, this is using LayoutGrid 0.10.0-dev.2.

I'm using LayoutGrid + OrientationBuilder to adjust rows/cols based on orientation. When Flutter rebuilds the widget upon an orientation change, I see the following crash:
```
════════ Exception caught by rendering library ═════════════════════════════════════════════════════
The following RangeError was thrown during performLayout():
RangeError (index): Invalid value: Not in range 0..1, inclusive: 2

The relevant error-causing widget was: 
  LayoutGrid file:///Users/adam/src/flutter_app/lib/main.dart:78:18
When the exception was thrown, this was the stack: 
#0      List.[] (dart:core-patch/array.dart:16:52)
#1      GridSizingInfo.offsetForArea (package:flutter_layout_grid/src/rendering/layout_grid.dart:779:38)
#2      RenderLayoutGrid.performLayout (package:flutter_layout_grid/src/rendering/layout_grid.dart:237:38)
#3      RenderObject.layout (package:flutter/src/rendering/object.dart:1767:7)
#4      _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:246:13)
```

Everything works fine in the new orientation after re-starting the app but if orientation is changed again the same error is thrown.

Here is a minimal example reproducing the issue:
```
class _MyHomePageState extends State<MyHomePage> {
  final List<Color> colors = [
    Colors.red,
    Colors.blue,
    Colors.green,
    Colors.purple,
    Colors.yellow,
    Colors.tealAccent,
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: _buildBody(),
    );
  }

  Widget _buildBody() {
    return OrientationBuilder(
      builder: (BuildContext context, Orientation orientation) {
        final isPortrait = orientation == Orientation.portrait;
        final rows = isPortrait ? 3 : 2;
        final cols = isPortrait ? 2 : 3;
        return Container(
          child: LayoutGrid(
            templateRowSizes:
                List<TrackSize>.generate(rows, (i) => FlexibleTrackSize(1)),
            templateColumnSizes:
                List<TrackSize>.generate(cols, (i) => FlexibleTrackSize(1)),
            children: List<Widget>.generate(6, (index) => _cellWidget(index)),
          ),
        );
      },
    );
  }

  Widget _cellWidget(int num) {
    return Container(
      color: colors[num],
      child: Center(child: Text('$num')),
    );
  }
}
```

The output rendered looks like this.

Working (portrait):
![Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-21 at 14 07 01](https://user-images.githubusercontent.com/869261/79824358-773f4b00-83d9-11ea-93d6-494249aefcfe.png)

Broken after orientation change (landscape):
![Simulator Screen Shot - iPhone 11 Pro Max - 2020-04-21 at 14 07 06](https://user-images.githubusercontent.com/869261/79824383-81f9e000-83d9-11ea-9f43-9e9e4e710ab9.png)

**Comment by shyndman** [comment]
Hey @adamkoch,

Sorry, I'm a little behind. Just saw this. I'll take a look tomorrow first thing.

**Comment by shyndman** [comment]
ok, I have successfully repro'd.

**Comment by shyndman** [comment]
Done!

Fix published as 0.10.1-dev.0

**Comment by adamkoch** [comment]
Awesome, thank you! Working well now 👍
